### PR TITLE
chore (deps): Update geth version to v1.10.25

### PR DESCRIPTION
**Pull request by ramacarlucho**
Closes: #XXX

## Description

Updated geth to `v1.10.25`. Implemented behaviour for the following changes: 
- Return ChainId for transactions [25155](https://github.com/ethereum/go-ethereum/pull/25155)
- Check ChainId on SendTransaction [25157](https://github.com/ethereum/go-ethereum/pull/25157)
- Added a new param for tracers [25430](https://github.com/ethereum/go-ethereum/pull/25430)

Example `debug_traceTransaction` request
```json
{
  "jsonrpc": "2.0",
  "method": "debug_traceTransaction",
  "params": [
    "0xa7da1489cdf8a23ff745b89346f454ede4e56a73a96ed95badee29cce1168401",
    {
      "tracer": "callTracer",
      "tracerConfig": "{ 'onlyTopCall': true }" 
    }
  ],
  "id": 1
}
```

Fixed some bugs in the integration tests suite. 
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/ibc-go/v5 v5.0.0
 	github.com/davecgh/go-spew v1.1.1
-	github.com/ethereum/go-ethereum v1.10.19
+	github.com/ethereum/go-ethereum v1.10.23
```
Comment (fedekunze) [inline go.mod:14]: can we update to the latest version (`v1.10.25`)

File: tests/integration_tests/test_pruned_node.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def test_pruned_node(pruned):
     test basic json-rpc apis works in pruned node
     """
     w3 = pruned.w3
-    erc20 = deploy_contract(
+    erc20,_ = deploy_contract(
```
Comment (fedekunze) [inline tests/integration_tests/test_pruned_node.py:37]:
```suggestion
    erc20, _ = deploy_contract(
```

File: tests/integration_tests/test_upgrade.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def test_cosmovisor_upgrade(custom_ethermint: Ethermint):
     print("upgrade height", target_height)
 
     w3 = custom_ethermint.w3
-    contract = deploy_contract(w3, CONTRACTS["TestERC20A"])
+    contract,_ = deploy_contract(w3, CONTRACTS["TestERC20A"])
```
Comment (fedekunze) [inline tests/integration_tests/test_upgrade.py:100]:
```suggestion
    contract, _ = deploy_contract(w3, CONTRACTS["TestERC20A"])
```

File: go.mod [hunk 4]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/ibc-go/v5 v5.0.1
 	github.com/davecgh/go-spew v1.1.1
-	github.com/ethereum/go-ethereum v1.10.19
+	github.com/ethereum/go-ethereum v1.10.25
```
Comment (fedekunze) [inline go.mod:14]:
```suggestion
	github.com/ethereum/go-ethereum v1.10.26
```

**Review by facs95 — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (facs95) [reply go.mod:15]: Lets do this in another PR cause this one has been stale for a long time. Will create a ticket for it

**Pull request merged**